docker-images/build.sh

```diff
@@ -1,57 +1,103 @@
 #!/usr/bin/env bash
 set -e
 
+# Image directories
 base_images="base"
 java_images="operator"
 kafka_images="kafka test-client"
 
-# Kafka versions
-function load_checksums {
-    declare -Ag checksums
-    while read line; do
-        checksums[$(echo "$line" | cut -d ' ' -f 1)]=$(echo "${line,,}" | cut -d ' ' -f 2)
-    done < <(sed -E -e '/^(#.*|[[:space:]]*)$/d' -e 's/^([0-9.]+)[[:space:]]+.*[[:space:]]+([[:alnum:]]+)[[:space:]]+.*$/\1 \2/g' ../kafka-versions)
+function dependency_check { 
+
+    # Check for bash >= 4
+    if [ -z ${BASH_VERSION+x} ]
+    then
+        # For some reason the bash version is not set so we set is ourselves
+        BASH_VERSION=$(/usr/bin/env bash --version |\
+                       grep -o '[0-9]\+\(\.[0-9]\+\)*' |\
+                       head -n 1 | cut -c 1)
```

Comment:
[`$BASH_VERSINFO`](http://www.tldp.org/LDP/abs/html/internalvariables.html) is probably what you're looking for, and if it's unset I would just exit with some error